src/services/words.ts

```diff
@@ -27,14 +30,28 @@ const getByLanguageAndUser = async function(languageId: string, userId: number):
   return result.rows.map((dbItem: WordDB) => convertWordTypes(dbItem));
 };
 
-
-const getUserwordsInText = async function(userId: number, textId: number, simple: boolean = true): Promise<Array<Word>> {
-  const result: QueryResult = await wordData.getUserwordsInText(userId, textId, simple);
-
-  return result.rows.map((dbItem: WordDB) => convertWordTypes(dbItem));
+const getUserwordsInText = async function(userId: number, textId: number, simple: boolean = true): Promise<Array<WordWithTransContextStatusTypes>> {
+  const wordsResult: QueryResult = await wordData.getUserwordsInText(userId, textId, simple);
+  const newWordsResult = wordsResult.rows;
+
+  console.log(newWordsResult);
+
+  await Promise.all(newWordsResult.map(async (item) => {
+    const translations = await translationData.getByWord(item.word, userId);
+    const status = await translationData.getStatusByWordByUser(item.id, userId);
```

Comment:
Use the existing `getStatus` function from `data-access/words.ts`.

Reply:
Is it possible to get the words, translations and statuses for all words in a text using one query? I imagine it would be something like the `getUserwordsInText` query but with joins for translations and statuses. I think (I may be wrong) that this is an example of N + 1 queries that we learned about in LS.

https://stackoverflow.com/questions/97197/what-is-the-n1-selects-problem-in-orm-object-relational-mapping

Reply:
You can use `array_agg` and `JOIN`s. I just tried it.